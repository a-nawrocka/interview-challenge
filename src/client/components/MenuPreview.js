import React from "react";
import Dish from "./Dish";

const MenuPreview = (props) => {
	return (
		<div className="col-8">
			<h2>Menu preview</h2>
			<ul className="menu-preview">
				{props.selectedDishes.map((dish, index) => (
					<Dish key={index} dish={dish} onRemoved={props.onDishRemoved} />
				))}
			</ul>
		</div>
	);
};

export default MenuPreview;
