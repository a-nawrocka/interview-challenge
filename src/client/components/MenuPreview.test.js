import React from "react";
import renderer from "react-test-renderer";
import MenuPreview from "./MenuPreview";

describe("Given valid array of selected dish objects", () => {
	it("displays list of selected dishes", () => {
		const expectedDishes = [
			{
				id: 1,
				name: "dish one",
				dietaries: ["v", "ve", "df"],
			},
			{
				id: 2,
				name: "dish two",
				dietaries: ["rsf", "n!"],
			},
		];

		const component = renderer.create(<MenuPreview selectedDishes={expectedDishes} onDishRemoved={() => {}} />);

		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
