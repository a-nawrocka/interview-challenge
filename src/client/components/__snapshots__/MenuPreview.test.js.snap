// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given valid array of selected dish objects displays list of selected dishes 1`] = `
<div
  className="col-8"
>
  <h2>
    Menu preview
  </h2>
  <ul
    className="menu-preview"
  >
    <li
      className="item"
    >
      <h2>
        dish one
      </h2>
      <p>
        <span
          className="dietary"
        >
          v
        </span>
        <span
          className="dietary"
        >
          ve
        </span>
        <span
          className="dietary"
        >
          df
        </span>
      </p>
      <button
        className="remove-item"
        onClick={[Function]}
      >
        x
      </button>
    </li>
    <li
      className="item"
    >
      <h2>
        dish two
      </h2>
      <p>
        <span
          className="dietary"
        >
          rsf
        </span>
        <span
          className="dietary"
        >
          n!
        </span>
      </p>
      <button
        className="remove-item"
        onClick={[Function]}
      >
        x
      </button>
    </li>
  </ul>
</div>
`;
